### PR TITLE
10653 – Add outline and cursor on geography hover

### PR DESCRIPTION
**Pull request by bmarchena**
### Summary
Adds highlighted outline around geographies on hover. Changes cursor when hovering over valid geographies.
~~- Cursor still needs to be added~~
~~- There is still an outstanding issue regarding performance.~~

#### Tasks/Bug Numbers
 - Fixes AB[#10653](https://dev.azure.com/NYCPlanning/ITD/_sprints/taskboard/Open%20Source%20Engineering/ITD/2022/Q4/Sprint%20U?workitem=10653)

https://user-images.githubusercontent.com/43344288/195644028-642b7c4b-e29a-434e-bfbb-f83821a0833a.mov

### CODE REVIEW

File: src/components/Map/Map.tsx [hunk 1]
```diff
@@ -12,37 +12,21 @@ import { useWindowWidth } from "@react-hook/window-size";
...
-    };
-  };
-}
-
-interface MapProps extends DeckProps {
-  hoverInfo: HoverType | null;
-}
-
-export const Map = ({ layers, parent, hoverInfo }: MapProps) => {
+type DeckProps = DeckGLProps<"parent">;
```
Comment (TylerMatteo) [inline src/components/Map/Map.tsx:23]: This is a little nitpicky but can you change this back to being called `MapProps`? It's a common convention in TS React to name your props types like "NameOfComponentProps"

File: src/components/Map/Map.tsx [hunk 2]
```diff
@@ -12,37 +12,21 @@ import { useWindowWidth } from "@react-hook/window-size";
...
+
+export const Map = ({ parent }: DeckProps) => {
+  const [hoverInfo, setHoverInfo] = useState<any | null>(null);
```
Comment (TylerMatteo) [inline src/components/Map/Map.tsx:26]: I'm fine with leaving this as `any` for now if you're having hard time typing it but just curious why this change? Could you use the `HoverType` interface that you got rid of here?

**Review by TylerMatteo — CHANGES_REQUESTED** [review]
Just one change I'd like to make but otherwise, lgtm! I tested the preview deployment and the performance seems to be at least as good as production.

File: src/components/Map/Map.tsx (hunk 2 above)
Comment (bmarchena) [reply src/components/Map/Map.tsx:26]: I tried that originally but got a funky type mismatch error because of the Dispatch action
Screen Shot 2022-10-17 at 11 13 41 AM

Comment (TylerMatteo) [reply src/components/Map/Map.tsx:26]: I see. All good. I'll take a crack at refactoring this after your merge.

**Review by TylerMatteo — APPROVED** [review]
